### PR TITLE
Retrieve bitcoin's mempool-min-fee

### PR DESCRIPTION
**Pull request by t-bast**
* Retrieve the minimum fee required to enter our mempool
* Add a metric to track this value
* Remove fallback fee providers when Bitcoin Core is used
* Add a metric to track feerate retrieval errors

For fallback fee providers that don't have any mempool information, we default to the 1008 block target. It doesn't matter as these fee providers will not be used anyway.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala [hunk 1]
```diff
@@ -244,7 +245,7 @@ class Setup(datadir: File,
         case (Block.RegtestGenesisBlock.hash, _) =>
           new FallbackFeeProvider(new ConstantFeeProvider(defaultFeerates) :: Nil, minFeeratePerByte)
         case (_, Bitcoind(bitcoinClient)) =>
-          new FallbackFeeProvider(new SmoothFeeProvider(new BitcoinCoreFeeProvider(bitcoinClient, defaultFeerates), smoothFeerateWindow) :: new SmoothFeeProvider(new BitgoFeeProvider(nodeParams.chainHash, readTimeout), smoothFeerateWindow) :: new SmoothFeeProvider(new EarnDotComFeeProvider(readTimeout), smoothFeerateWindow) :: Nil, minFeeratePerByte) // order matters!
+          new FallbackFeeProvider(new SmoothFeeProvider(new BitcoinCoreFeeProvider(bitcoinClient, defaultFeerates), smoothFeerateWindow) :: Nil, minFeeratePerByte)
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala:248]: I think we can remove the FallbackFeeProvider in that case.

File: eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/FeeProvider.scala [hunk 2]
```diff
@@ -102,8 +102,8 @@ object FeeratePerKw {
 }
 
 /** Fee rates in satoshi-per-kilo-bytes (1 kb = 1000 bytes). */
-case class FeeratesPerKB(block_1: FeeratePerKB, blocks_2: FeeratePerKB, blocks_6: FeeratePerKB, blocks_12: FeeratePerKB, blocks_36: FeeratePerKB, blocks_72: FeeratePerKB, blocks_144: FeeratePerKB, blocks_1008: FeeratePerKB) {
-  require(block_1.feerate > 0.sat && blocks_2.feerate > 0.sat && blocks_6.feerate > 0.sat && blocks_12.feerate > 0.sat && blocks_36.feerate > 0.sat && blocks_72.feerate > 0.sat && blocks_144.feerate > 0.sat && blocks_1008.feerate > 0.sat, "all feerates must be strictly greater than 0")
+case class FeeratesPerKB(mempoolMinFee: FeeratePerKB, block_1: FeeratePerKB, blocks_2: FeeratePerKB, blocks_6: FeeratePerKB, blocks_12: FeeratePerKB, blocks_36: FeeratePerKB, blocks_72: FeeratePerKB, blocks_144: FeeratePerKB, blocks_1008: FeeratePerKB) {
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/FeeProvider.scala:105]: Maybe you can squeeze a comment somewhere explaining what `mempoolMinFee` is and why it's ok to set it to `blocks_1008` if the provider doesn't offer finer-grained data?

**Review by pm47 — COMMENTED** [review]
Nits, otherwise LGTM.

File: eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala (hunk 1 above)
Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala:248]: In fact we can't, it's that `FallbackFeeProvider` that enforces the minimum feerate...
Once we've removed bitgo / earndotcom, we'll be able to clean this up a bit, but let's keep it for now?

File: eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/FeeProvider.scala (hunk 2 above)
Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/FeeProvider.scala:109]: Done in https://github.com/ACINQ/eclair/pull/1608/commits/ce14f3742f74fd134762673a40b99225b198cd2f

**Review by pm47 — APPROVED** [review]

**Pull request merged**